corehq/motech/repeaters/tasks.py

```diff
@@ -139,43 +178,19 @@ def retry_process_repeat_record(repeat_record_id, domain):
     NOTE: Keep separate from process_repeat_record for monitoring purposes
     Domain is present here for domain tagging in datadog
     """
-    _process_repeat_record(RepeatRecord.objects.get(id=repeat_record_id))
+    return _process_repeat_record(RepeatRecord.objects.get(id=repeat_record_id))
 
 
 def _process_repeat_record(repeat_record):
-    if repeat_record.state == State.Cancelled:
-        return
-
-    if not domain_can_forward(repeat_record.domain) or repeat_record.exceeded_max_retries:
-        # When creating repeat records, we check if a domain can forward so
-        # we should never have a repeat record associated with a domain that
-        # cannot forward, but this is just to be sure
-        repeat_record.cancel()
-        repeat_record.save()
-        return
-
-    if repeat_record.repeater.is_deleted:
-        repeat_record.cancel()
-        repeat_record.save()
-        return
-
+    state_or_none = None
     try:
-        if repeat_record.repeater.is_paused or toggles.PAUSE_DATA_FORWARDING.enabled(repeat_record.domain):
-            # postpone repeat record by MAX_RETRY_WAIT so that it is not fetched
-            # in the next check to process repeat records, which helps to avoid
-            # clogging the queue
-            repeat_record.postpone_by(MAX_RETRY_WAIT)
-        elif rate_limit_repeater(repeat_record.domain):
-            # Spread retries evenly over the range defined by RATE_LIMITER_DELAY_RANGE
-            # with the intent of avoiding clumping and spreading load
-            repeat_record.postpone_by(random.uniform(*RATE_LIMITER_DELAY_RANGE))
-        elif repeat_record.is_queued():
```

Comment:
should `queued` state still be checked?

Reply:
`is_queued()` just checks `self.state == State.Pending or self.state == State.Fail`, and we know this must be true.

Reply:
Unless it was processed by another task before it got here right?